clients/pkg/promtail/promtail.go

```diff
@@ -61,37 +72,62 @@ func New(cfg config.Config, metrics *client.Metrics, dryRun bool, opts ...Option
 		}
 		o(promtail)
 	}
+	err := promtail.reloadConfig(&cfg)
+	if err != nil {
+		return nil, err
+	}
+	server, err := server.New(cfg.ServerConfig, promtail.logger, promtail.targetManagers, cfg.String())
+	if err != nil {
+		return nil, err
+	}
+	promtail.server = server
+	promtail.newConfig = newConfig
 
-	cfg.Setup(promtail.logger)
+	return promtail, nil
+}
+
+func (p *Promtail) reloadConfig(cfg *config.Config) error {
+	level.Info(p.logger).Log("msg", "Loading configuration file")
```

Comment:
Let's be clear here about what is happening
We should also consider logging the path of the config file

```suggestion
	level.Info(p.logger).Log("msg", "Reloading configuration file")
```